mlflow/tensorflow.py

```diff
@@ -333,3 +344,238 @@ def predict(self, df):
             raw_preds = self.tf_sess.run(self.output_tensors, feed_dict=feed_dict)
             pred_dict = {column_name: values.ravel() for column_name, values in raw_preds.items()}
             return pandas.DataFrame(data=pred_dict)
+
+
+class __MLflowTfKerasCallback(Callback):
+    """
+        Callback for auto-logging parameters (we rely on TensorBoard for metrics).
+        Records model structural information as params after training finishes.
+    """
+    def __init__(self):
+        if mlflow.active_run() is None:
+            mlflow.start_run()
+
+    def __enter__(self):
+        pass
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        pass
+
+    def on_epoch_end(self, epoch, logs=None):
+        pass
+
+    def on_train_end(self, logs=None):  # pylint: disable=unused-argument
+        opt = self.model.optimizer
+        if hasattr(opt, 'optimizer'):
+            opt = opt.optimizer
+        mlflow.log_param('optimizer_name', type(opt).__name__)
+        if hasattr(opt, '_lr'):
+            lr = opt._lr if type(opt._lr) is float else tensorflow.keras.backend.eval(opt._lr)
+            mlflow.log_param('learning_rate', lr)
+        if hasattr(opt, '_epsilon'):
+            epsilon = opt._epsilon if type(opt._epsilon) is float \
+                else tensorflow.keras.backend.eval(opt._epsilon)
+            mlflow.log_param('epsilon', epsilon)
+        l = []
+        self.model.summary(print_fn=l.append)
+        summary = '\n'.join(l)
+        mlflow.set_tag('summary', summary)
+        # TODO: Fix for keras log_model not saving TF optimizers
```

Comment:
Nit: can we remove this TODO?

Reply:
Yeah, looks like it snuck back in when I reset to fix a commit issue. Good catch